src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -356,6 +437,35 @@ export class SavedObjectsClient {
       return Promise.reject(new Error('body not permitted for GET requests'));
     }
 
-    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) });
+    return this.http.fetch(path, { method, query, body: JSON.stringify(body) }).catch(err => {
+      // To ensure we don't break backwards compatibility, we keep the old
+      // kfetch error format of `{res: {status: number}}` whereas `http.fetch`
+      // uses `{response: {status: number}}`
+
+      const kfetchError = Object.assign(err, { res: err.response });
+      return Promise.reject(kfetchError);
+    });
   }
 }
+
+/**
+ * Returns a new object with the own properties of `obj`, but the
+ * keys renamed according to the `keysMap`.
+ *
+ * If a key is not found in `keysMap` it is omitted from the result.
+ *
+ * @param keysMap - a map of the form `{oldKey: newKey}`
+ * @param obj - the object whose own properties will be renamed
+ */
+const pickAndRenameKeys = <T extends Record<string, any>, U extends Record<string, any>>(
```

Comment:
optional nit: usually `and` in name is a smell. does make sense to split the function in 2 - filterKeys and mapKeys?